src/epplibwrapper/tests/test_client.py

```diff
@@ -255,3 +261,101 @@ def side_effect(*args, **kwargs):
             mock_close.assert_called_once()
             # send() is called 5 times: send(login), send(command) fail, send(logout), send(login), send(command)
             self.assertEquals(mock_send.call_count, 5)
+
+    def test_send_command_close_failure_recovers(self):
+        """Test when the .close on a connection fails and a .send follows suit.
+        Flow:
+        Initialization succeeds
+        Send command fails (with 2400 code) prompting retry
+        Client closes and re-initializes, and command succeeds"""
+
+        expected_result = {
```

Comment:
Do all these functions and this large expected result have to be in this test? Can they move outside of it? This make the test a bit hard to follow.